### ISSUE TITLE
Dotnet core is not supported in xunit.runner.visualstudio version 2.2.0.beta4-build119

### ISSUE THREAD

**Issue by akazemis** [xunit/xunit#1032]
Given the following Project.json file (dotnet core project):
``` json
{
  "version": "1.0.0-*",
  "testRunner": "xunit",
  "dependencies": {
    "xunit": "2.2.0-beta4-build3444",
    "Moq": "4.6.38-alpha",
    "Ixup.Models": "1.0.0-*",
    "FluentAssertions": "4.15.0",
    "dotnet-test-xunit": "2.2.0-preview2-build1029",
    "xunit.runner.visualstudio": "2.2.0-beta4-build1194"
  },

  "frameworks": {
    "netcoreapp1.0": {
      "imports": "dnxcore50",
      "dependencies": {
        "Microsoft.NETCore.App": {
          "type": "platform",
          "version": "1.0.0"
        }
      }
    }
  }
}
```

I'm getting this error while restoring my NuGet packages :

> Errors in E:\Dev\IXUP_App\IxupCore\tests\Ixup.Models.Tests\Ixup.Models.Tests.xproj
>     Package Microsoft.TestPlatform.ObjectModel 11.0.0 is not compatible with netcoreapp1.0 (.NETCoreApp,Version=v1.0). Package Microsoft.TestPlatform.ObjectModel 11.0.0 supports:
>       - net35 (.NETFramework,Version=v3.5)
>       - portable-net45+win8+wp8+wpa81 (.NETPortable,Version=v0.0,Profile=Profile259)
>     One or more packages are incompatible with .NETCoreApp,Version=v1.0.

Seemingly this package doesn't support dotnet core, whereas it's explicitly mentioned in the NuGet package description:

```
Visual Studio 2012+ Test Explorer runner for the xUnit.net framework. Capable of running xUnit.net v1.9.2 and v2.0+ tests.

Supported platforms:
- Desktop .NET 2.0+
- .NET Core (including ASP.NET Core) 1.0+
- Universal Windows App 10+
- Windows App 8.1
- Windows Phone App 8.1
```

**Comment by BenjaminAbt** [comment]
I get the same error for .NET Core 1.1. My setup:

```
{
  "version": "1.0.0-*",

  "testRunner": "xunit",

  "dependencies": {
    "NETStandard.Library": "1.6.1",
    "xunit": "2.2.0-beta4-build3444",
    "xunit.runner.visualstudio": "2.2.0-beta4-build1194"
  },

  "frameworks": {
    "netcoreapp1.1": {
      "dependencies": {
        "Microsoft.NETCore.App": {
          "type": "platform",
          "version": "1.1.0"
        }
      }
    }
  }
}
```

dotnet info:

```
C:\Users\Ben>dotnet --info
.NET Command Line Tools (1.0.0-preview2-1-003177)

Product Information:
 Version:            1.0.0-preview2-1-003177
 Commit SHA-1 hash:  a2df9c2576

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Visual Studio 2015 Update 3
```

**Comment by charlenni** [comment]
I have the same problems trying to test a portable library, which uses NetStandard 1.6.1. Tried it allready like described on the webpage, but it didn't worked.

**Comment by codito** [comment]
The `preview2-*` version of dotnet-cli toolset supports `dotnet-test-xunit`, `xunit.runner.visualstudio` is not compatible.

The newer (`preview3-*`) version of dotnet-cli toolset supports `xunit.runner.visualstudio` based test projects. It is not compatible with dotnet-test-xunit. Note that this toolset requires csproj based dotnet projects.

**Comment by onovotny** [comment]
Closing as @codito's answer is correct.

**Comment by Pastafarian** [comment]
@onovotny @codito - can you confirm that xunit.net is not compatible with the current version of dotnet core in Visual Studio 2015 and requires 2017 RC1 to work?

**Comment by onovotny** [comment]
xunit does work with VS 2015 and the Preview 2.1 tooling, but you need the right versions. You can see the versions here: http://xunit.github.io/docs/getting-started-dotnet-core.html. For VS 2015/project.json, you'll need to use `dotnet-test-xunit`.

**Comment by thj-dk** [comment]
I'm confused by all these versions... Does xUnit work with VS 2017 RC? I've installed xunit.runner.visualstudio v. 2.2.0-beta4-build3444 in my .NET Standard 1.5 class library. The dotnet-cli version is 1.0.0-preview4-004233 (if that has any influence).
But no tests are discovered by VS or ReSharper 2016.3. Any ideas?

**Comment by bradwilson** [comment]
@thj-dk wrote:

> in my .NET Standard 1.5 class library

The xUnit.net Visual Studio plugin only supports the following platforms:

- Desktop: `net45` and later
- Windows Phone: `wpa81` (being removed in 2.2)
- Windows Store: `win81` (being removed in 2.2)
- Universal Windows: `uwp10.0` and later
- .NET Core: `netcoreapp1.0` and later (starting with 2.2 beta 4)

It does not support PCLs or `netstandard` DLLs. Why? Because those have no associated platform; they are, by definition, platform agnostic. The Visual Studio plugin needs to know what platform the tests are intended to run on. The solution to your problem is to target one ore more specific platforms.

**Comment by thj-dk** [comment]
@bradwilson Thanks alot. That makes sense. So the solution would be to create a net45/netcoreapp1.0 class library to test a netstandard class library, i guess?

**Comment by bradwilson** [comment]
Yes. Multi-targeting your unit tests for all desired test platforms is what makes sense.

The integration for multi-target projects with the test runner inside Visual Studio is... somewhere between wonky and broken. In the meantime, using separate class library projects with some form of shared source works better.

**Comment by BalassaMarton** [comment]
`xunit.runners.visualstudio` also fails to install on Visual Studio 2017 RTM. Steps to reproduce:
1. Create New Class Library (.NET Core)
2. Install NuGet package: `xunit` `2.2.0`
3. Install NuGet package: `xunit.runners.visualstudio` `2.2.0`
Install fails with the following output:

```
Package Microsoft.TestPlatform.ObjectModel 11.0.0 is not compatible with netcoreapp1.0 (.NETCoreApp,Version=v1.0). Package Microsoft.TestPlatform.ObjectModel 11.0.0 supports:
  - net35 (.NETFramework,Version=v3.5)
  - portable-net45+win8+wp8+wpa81 (.NETPortable,Version=v0.0,Profile=Profile259)
One or more packages are incompatible with .NETCoreApp,Version=v1.0.
Package restore failed. Rolling back package changes for 'ClassLibrary1'.
```

**Comment by BalassaMarton** [comment]
Found the solution here:
https://docs.microsoft.com/en-us/dotnet/articles/core/testing/unit-testing-with-dotnet-test
TL;DR: install the `Microsoft.NET.Test.Sdk` NuGet first.
My guess is that `xunit.runners.visualstudio` references a previous version of `Microsoft.TestPlatform.ObjectModel` that does not target .NET Core. Adding the `Microsoft.NET.Test.Sdk` 
package solves the problem because it references the `15.0.0` version.

**Comment by epsitec** [comment]
For those who still cannot manage to get `xUnit 2.2` and Visual Studio 2017 to work together, you might have some outdated SDK previews getting in the way. My **Test Explorer** window remained empty, no matter what I did, until I finally uninstalled all x86 SDKs, and removed the the previews found in `C:\Program Files\dotnet\sdk` to only keep `1.0.0` and `1.0.1`.

I blogged about [what I did](http://code.fitness/post/2017/03/dotnet-core-vs2017-tests-not-working.html)
to get things working.

**Comment by raykrow** [comment]
@epsitec's solution worked for me (kindof). I started by only deleting what I had to but it still didn't work. After removing VS2017, VS2015 and ALL dotnet core and **only** reinstalling VS2017 the problem was solved.

http://stackoverflow.com/questions/42861930/unit-tests-not-discovered-in-visual-studio-2017/43158783#43158783

**Comment by YLPeng** [comment]
@bradwilson  Thanks for the comments. I'm trying to make a .NetStandard library, and according to the comments above, I should create unit tests in a separate project such as .NetCore. My question is can both .NetStandard and .NetCore be in the same solution? Currently I put both in the same solution but it seems I cannot reference .NetStandard library from .NetCore(unit test) class.

**Comment by bradwilson** [comment]
> My question is can both .NetStandard and .NetCore be in the same solution?

Yes.

> it seems I cannot reference .NetStandard library from .NetCore(unit test) class.

Can you provide a repro? This should just work. .NET Core < v2.0 can use any version of .NET Standard <= v1.6 (and neither .NET Core 2.0 nor .NET Standard 2.0 are supported by current tooling).

**Comment by ashokprecise** [comment]
Hi,
I am using VisulaStudio2015 Update3 in my .netcore project i have followed all instructions that you provide here. but no luck i cant able to discovered the test result is 0. i don't know what i miss there.

**Comment by YLPeng** [comment]
@bradwilson  Thanks so much for getting me back. Yeah it worked, it was a mistake on my part.
@ashokprecise Why not try with VS2017 since it's officially released.
